supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/os/SigningProvider.kt

```diff
@@ -0,0 +1,211 @@
+     * Omit to default to the only supported padding.
+     *
+     * If the key stored in hardware supports multiple padding modes, you need to explicitly specify the digest to use.
+     * (By default, hardware keys are configured to only support a single digest.)
+     *
+     * @see SigningKeyConfiguration.RSAConfiguration.paddings
+     */
+    lateinit var padding: RSAPadding
+    internal val paddingSpecified get() = this::padding.isInitialized
+
+
+}
+open class SignerConfiguration internal constructor(): DSL.Data() {
+    /** Algorithm-specific configuration for a returned ECDSA signer. Ignored for RSA keys. */
+    open val ec = childOrDefault(::ECSignerConfiguration)
+    /** Algorithm-specific configuration for a returned RSA signer. Ignored for ECDSA keys. */
+    open val rsa = childOrDefault(::RSASignerConfiguration)
+}
+
+open class UnlockPromptConfiguration: DSL.Data() {
+
+    internal val _message = Stackable<String>()
+    /** The prompt message to show to the user when asking for unlock */
+    var message by _message
+
+    internal val _cancelText = Stackable<String>()
+    /** The message to show on the cancellation button */
+    var cancelText by _cancelText
+
+    companion object {
+        const val defaultMessage = "Please authorize cryptographic signature"
+        const val defaultCancelText = "Cancel"
+    }
+}
+open class PlatformSignerConfigurationBase internal constructor(): SignerConfiguration() {
+    /** Configure the authorization prompt that will be shown to the user. */
+    open val unlockPrompt = childOrDefault(::UnlockPromptConfiguration)
+}
+
+open class PlatformSigningProviderSignerSigningConfigurationBase internal constructor(): DSL.Data() {
+    open val unlockPrompt = childOrDefault(::UnlockPromptConfiguration)
+}
+
+interface PlatformSigningProviderSigner<SigningConfiguration: PlatformSigningProviderSignerSigningConfigurationBase>
+    : Signer.WithAlias {
+
+    suspend fun trySetupUninterruptedSigning(configure: DSLConfigureFn<SigningConfiguration> = null) : KmmResult<Unit> = KmmResult.success(Unit)
+    override suspend fun trySetupUninterruptedSigning() = trySetupUninterruptedSigning(null)
+
+    suspend fun sign(data: SignatureInput, configure: DSLConfigureFn<SigningConfiguration> = null) : KmmResult<CryptoSignature>
+    suspend fun sign(data: ByteArray, configure: DSLConfigureFn<SigningConfiguration> = null) =
+        sign(SignatureInput(data), configure)
+    suspend fun sign(data: Sequence<ByteArray>, configure: DSLConfigureFn<SigningConfiguration> = null) =
+        sign(SignatureInput(data), configure)
+    override suspend fun sign(data: SignatureInput) = sign(data, null)
+    override suspend fun sign(data: ByteArray) = sign(SignatureInput(data), null)
+    override suspend fun sign(data: Sequence<ByteArray>) = sign(SignatureInput(data), null)
+}
+
+open class PlatformSigningProviderConfigurationBase internal constructor(): DSL.Data()
+// BLOCKED BY KT-71036
```

Comment:
then it is ok to keep the commented-out code

Reply:
#107